lib/galaxy/datatypes/flow.py

```diff
@@ -0,0 +1,435 @@
+
+    def sniff(self, filename):
+        """
+        Checking if the file is a FlowSOM R object.
+
+        For this to work, need to have install checkFlowSOM.R via bioconda
+        conda install ig-checkflowtypes
+        """
+        try:
+            rscript = 'checkFlowSOM.R'
+            fcs_check = subprocess.check_output([rscript, filename])
+            if re.search('TRUE', str(fcs_check)):
+                return True
+            else:
+                return False
+        except Exception:
+            False
+
+    def get_mime(self):
+        """Returns the mime type of the datatype"""
+        return 'application/octet-stream'
+
+
+Binary.register_sniffable_binary_format('fsom', 'fsom', FlowSOM)
+
+
+class FlowSet(Binary):
+    """R Object containing flowSet saved with saveRDS"""
+    file_ext = 'flowset'
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = "Binary RDS flowSet file"
+            dataset.blurb = data.nice_size(dataset.get_size())
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disk'
+
+    def display_peek(self, dataset):
+        try:
+            return dataset.peek
+        except Exception:
+            return "Binary RDS flowSet (%s)" % (data.nice_size(dataset.get_size()))
+
+    def sniff(self, filename):
+        """
+        Checking if the file is a flowSet R object.
+
+        For this to work, need to have install checkFlowSet.R via bioconda
+        conda install ig-checkflowtypes
+        """
+        try:
+            rscript = 'checkFlowSet.R'
+            fcs_check = subprocess.check_output([rscript, filename])
+            if re.search('TRUE', str(fcs_check)):
+                return True
+            else:
+                return False
+        except Exception:
+            False
+
+    def get_mime(self):
+        """Returns the mime type of the datatype"""
+        return 'application/octet-stream'
+
+
+Binary.register_sniffable_binary_format('flowset', 'flowset', FlowSet)
+
+
+class FlowText(Tabular):
+    """Class describing an Flow Text file"""
+    file_ext = "flowtext"
+
+    def set_peek(self, dataset, is_multi_byte=False):
+        if not dataset.dataset.purged:
+            dataset.peek = "Text Flow file"
+            dataset.blurb = data.nice_size(dataset.get_size())
+        else:
+            dataset.peek = 'file does not exist'
+            dataset.blurb = 'file purged from disk'
+
+    def display_peek(self, dataset):
+        try:
+            return dataset.peek
+        except Exception:
+            return "Text Flow file (%s)" % (data.nice_size(dataset.get_size()))
+
+    def sniff(self, filename):
+        """Quick test on file formatting and values"""
+        with open(filename, "r") as f:
+            f.readline()
+            values = f.readline().strip().split("\t")
```

Comment:
This might be to generic to be sniffable.

Reply:
what do you mean by too generic here? could you elaborate please? thanks!

Reply:
I think the chance of false positive hits of the sniffer is to high. Essentially it returns `true` for all tsv files with only numbers in it.

Reply:
well, apparently a flowtext file seems to be just that. I could in turn ask pandas to read the table and ask for summaries on all columns, but it will be just that, numbers on every row besides the header. I'm guessing is not worth it for an sniffer.

Reply:
So should I just remove the sniffer method or the whole class and just add it as a sub datatype in the conf?